nixos/modules/services/hardware/udisks2.nix

```diff
@@ -21,23 +23,43 @@ with lib;
         '';
       };
 
+      mountAtMedia = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          By default, udisks2 mounts removable drives under the ACL
+          controlled directory /run/media/$USER/. If you wish to
+          mount to /media instead, enable this option.
+
+          This option will also clean /media at every boot.
+          Since /media, unlike /run, is not mounted by default as a
+          tmpfs, /media directory needs to be cleaned to get rid of
+          stale mountpoints.
+        '';
+      };
+
     };
 
   };
 
 
   ###### implementation
 
-  config = mkIf config.services.udisks2.enable {
+  config = mkIf cfg.enable {
 
     environment.systemPackages = [ pkgs.udisks2 ];
 
     services.dbus.packages = [ pkgs.udisks2 ];
 
-    systemd.tmpfiles.rules = [ "d /var/lib/udisks2 0755 root root -" ];
+    systemd.tmpfiles.rules = [ "d /var/lib/udisks2 0755 root root -" ]
+      ++ optional cfg.mountAtMedia "D! /media 0755 root root - ";
 
     services.udev.packages = [ pkgs.udisks2 ];
 
+    services.udev.extraRules = optionalString cfg.mountAtMedia ''
+      ENV{ID_FS_USAGE}=="filesystem", ENV{UDISKS_FILESYSTEM_SHARED}="1"
```

Comment:
Why not also `other|crypto` like in the Arch wiki?

Reply:
I actually took it from here:
https://sites.google.com/site/installationubuntu/home/ubuntu-20-04/change-mountpoint-of-usb-drives